### PR TITLE
support for alive2 overflow instrs

### PR DESCRIPTION
**Pull request by pranavk**

### CODE REVIEW

File: lib/Infer/AliveDriver.cpp [hunk 1]
```diff
@@ -530,12 +536,21 @@ std::string getUniqueName() {
...
   }
 
+  auto Ops = I->Ops;
+  if (souper::Inst::isOverflowIntrinsicMain(I->K)) {
+    Ops = Ops[0]->Ops;
+  }
+
   FunctionBuilder Builder(F);
-  for (auto &&Op : I->Ops) {
+  for (auto Op : Ops) {
```
Comment (regehr) [inline lib/Infer/AliveDriver.cpp]: plz explain why you chose to make a copy of each op instead of using rvalue references

**Comment by pranavk** [comment]
Oh, yeah, this can be changed now. I was doing a different approach first
and then changed to another approach. This removal of rvalue ref is remnant
of the previous approach. I will fix it

File: lib/Infer/AliveDriver.cpp (hunk 1 above)
Comment (pranavk) [reply lib/Infer/AliveDriver.cpp]: fixed it.

File: lib/Infer/AliveDriver.cpp [hunk 2]
```diff
@@ -530,12 +536,21 @@ std::string getUniqueName() {
 bool souper::AliveDriver::translateAndCache(const souper::Inst *I,
                                             IR::Function &F,
                                             Cache &ExprCache) {
+  if (souper::Inst::isOverflowIntrinsicSub(I->K)) {
```
Comment (regehr) [inline lib/Infer/AliveDriver.cpp]: it seems odd to special-case this one instruction, what's going on here?

Comment (pranavk) [reply lib/Infer/AliveDriver.cpp]: we are special casing out the SAddO instruction as they are not real instructions. The selected function checks if the given instruction kind is any of those 'AddO' type.

Comment (regehr) [reply lib/Infer/AliveDriver.cpp]: ok-- add a brief comment to this effect, that these are souper-internal things that we don't bother alive with

Comment (pranavk) [reply lib/Infer/AliveDriver.cpp]: done

**Comment by regehr** [comment]
LGTM, thanks!

**Pull request merged**